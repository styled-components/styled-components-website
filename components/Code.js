import styled from 'styled-components'
import rem from '../utils/rem'
import { lightGrey } from '../utils/colors'
import { monospace } from '../utils/fonts'

import '../utils/prismTemplateString'
import { Editor } from 'react-live'

const Code = styled.span`
  font-family: ${monospace};
  font-weight: 500;
`

export default Code
