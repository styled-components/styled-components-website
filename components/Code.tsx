import styled from 'styled-components';
import { monospace } from '../utils/fonts';

const Code = styled.code`
  border-radius: 3px;
  background: rgba(10, 10, 10, 0.1);
  display: inline-block;
  font-family: ${monospace};
  font-size: 90%;
  line-height: 1;
  padding: 0.3em 0.25em;
  text-decoration: inherit;
  vertical-align: baseline;
`;

export default Code;
