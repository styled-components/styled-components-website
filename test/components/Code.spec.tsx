import React from 'react';
import renderer from 'react-test-renderer';
import Code from '../../components/Code';

test('Code renders correctly', () => {
  const tree = renderer.create(<Code />).toJSON();

  expect(tree).toMatchSnapshot();
});
