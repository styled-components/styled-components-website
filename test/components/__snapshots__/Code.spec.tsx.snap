// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code renders correctly 1`] = `
.c0 {
  border-radius: 3px;
  background: rgba(10, 10, 10, 0.1);
  display: inline-block;
  font-family: dm,monospace;
  font-size: 90%;
  line-height: 1;
  padding: 0.3em 0.25em;
  text-decoration: inherit;
  vertical-align: baseline;
}

<code
  className="c0"
/>
`;
